source/ast/Scope.cpp

```diff
@@ -620,6 +620,16 @@ static bool canLookupByName(SymbolKind kind) {
     }
 }
 
+void Scope::nameMapEmplace(const Symbol* member, bool isElaborating) const {
```

Comment:
I don't think you need to factor this out; when you call this for the genblks you already know the name is not empty, it canBeLookedUpByName, and that there won't be any name conflicts.